builtin/logical/database/path_config_connection.go

```diff
@@ -394,6 +394,11 @@ func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 			return logical.ErrorResponse("error creating database object: %s", err), nil
 		}
 		config.ConnectionDetails = initResp.Config
+		// don't override a manually specified plugin version
+		// but if it is empty, then add in the self-reported version for informational purposes
+		if config.PluginVersion == "" {
+			config.PluginVersion = dbw.Version().Version
```

Comment:
I think this could cause issues the next time the config is updated if it's different to the version we did the initial lookup with, as next time we'll lookup the reported version and maybe get a different entry from the catalog. Could we call this method as we're registering the plugin, and either a) Use the reported version in the catalog if none was provided or b) Error if the reported version is different from the one provided? Then we shouldn't need to make any updates here as the catalog and reported versions will always match up.

It might be interesting to query the version from the plugin and validate that it's the version we think it should be or error out, though I can't think of any reason that would fail if we validate during plugin registration.

Reply:
It is definitely possible to add a check between `PluginVersion` and the self-reported version on registration.

I primarily added this because, if no version is specified on registration, it is otherwise impossible to tell what version of the plugin is running, and I wanted to expose this somewhere.

It doesn't even really make sense to have this information in the config at all, but there's no mount or auth table for database configs.